multi-agent/adapters/inbound/flask/identity.py

```
@@ -0,0 +1,160 @@
+                }), 401
+
+            if user_id not in admin_allowed_users:
+                return jsonify({
+                    "error": "Access denied: insufficient permissions",
+                    "error_type": "ACCESS_DENIED",
+                }), 403
+
+            return f(*args, **kwargs)
+
+        except Exception as e:
+            logger.error("Access control error", exc_info=True)
```

Ruff (0.15.15)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)